### PR TITLE
SPOI-8728 Json Parser

### PR DESCRIPTION
**Pull request by shubham-pathak22**

### CODE REVIEW

**Comment by shubham-pathak22** [comment]
@ashwinchandrap Could you review this ?

**Comment by shubham-pathak22** [comment]
@sandeepdeshmukh @chinmaykolhatkar Could you review this ?

File: tutorials/parser/pom.xml [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    </plugins>
+
+  </build>
+
+  <dependencies>
+    <!-- add your dependencies here -->
+    <dependency>
+      <groupId>org.apache.apex</groupId>
+      <artifactId>malhar-library</artifactId>
+      <version>3.4.0</version>
```
Comment (chinmaykolhatkar) [inline tutorials/parser/pom.xml]: Can you use variables instead?

File: tutorials/parser/src/main/java/com/datatorrent/tutorial/jsonparser/Application.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ConsoleOutputOperator pojo = dag.addOperator("Pojo", ConsoleOutputOperator.class);
+    ConsoleOutputOperator jsonObject = dag.addOperator("JsonObject", ConsoleOutputOperator.class);
+    ConsoleOutputOperator error = dag.addOperator("Error", ConsoleOutputOperator.class);
+
+    dag.addStream("json", generator.out, parser.in);
+    dag.addStream("pojo", parser.out, pojo.input);
+    dag.addStream("jsonObject", parser.parsedOutput, jsonObject.input);
+    dag.addStream("error", parser.err, error.input);
+    
+    dag.getMeta(parser).getMeta(parser.out).getAttributes().put(Context.PortContext.TUPLE_CLASS, Campaign.class);
```
Comment (chinmaykolhatkar) [inline tutorials/parser/src/main/java/com/datatorrent/tutorial/jsonparser/Application.java]: Can you set TUPLE_CLASS from properties.xml?

**Comment by chinmaykolhatkar** [comment]
@shubham-pathak22 Can you use the same application to showcase both JSONFormatter and JSONParser. So application might look like this:
JsonDataGenerator -> JSONParser -> JSONFrmatter -> ConsoleOutput.

**Comment by shubham-pathak22** [comment]
Incorporated review comments. JsonFormatter is still in PR review stage. 
https://github.com/apache/apex-malhar/pull/348

File: tutorials/parser/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+This project contains applications showcasing different **Parsers** and **Formatters** present in the Malhar Library. For all the apps, parameters are configuratble in META_INF/properties.xml.
```
Comment (chinmaykolhatkar) [inline tutorials/parser/README.md]: Knit: "configurable"

File: tutorials/parser/pom.xml [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+          <groupId>*</groupId>
+          <artifactId>*</artifactId>
+        </exclusion>
+      </exclusions>
+      -->
+    </dependency>
+    <dependency>
+      <groupId>org.apache.apex</groupId>
+      <artifactId>malhar-contrib</artifactId>
+      <version>3.5.0-SNAPSHOT</version>
```
Comment (chinmaykolhatkar) [inline tutorials/parser/pom.xml]: Can you create a variable for this? This is to ensure that further changes are easily and nothing is missed.

File: tutorials/parser/src/main/java/com/datatorrent/tutorial/jsonparser/Campaign.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+public class Campaign
+{
+  public int adId;
+  public String campaignName;
+  @JsonProperty("budget")
+  public double campaignBudget;
+  public boolean weatherTargeting;
+  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy/MM/dd")
+  public Date startDate;
+
```
Comment (chinmaykolhatkar) [inline tutorials/parser/src/main/java/com/datatorrent/tutorial/jsonparser/Campaign.java]: No getter/setter required for this? What happans when the variables are private? Can you have a proper POJO instead?

File: tutorials/parser/src/main/java/com/datatorrent/tutorial/jsonparser/Application.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import com.datatorrent.api.annotation.ApplicationAnnotation;
+import com.datatorrent.contrib.parser.JsonParser;
+import com.datatorrent.lib.formatter.JsonFormatter;
+import com.datatorrent.lib.io.ConsoleOutputOperator;
+
+@ApplicationAnnotation(name = "JsonProcessor")
+public class Application implements StreamingApplication
+{
+
+  public static final String COOL_DOWN_MILLIS = "dt.application.JsonProcessor.coolDownMillis";
```
Comment (chinmaykolhatkar) [inline tutorials/parser/src/main/java/com/datatorrent/tutorial/jsonparser/Application.java]: Where are these variables used? COOL_DOWN_MILLIS, MAX_THROUGHPUT, MIN_THROUGHOUT?

**Comment by shubham-pathak22** [comment]
Incorporated review comments.

**Pull request merged**